src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+            case 17:
+                *dListP = &D_06004950;
+                break;
+            case 29:
+                *dListP = &D_060049E8;
+                break;
+            case 5:
+                *dListP = &D_06003FB0;
+                break;
+            case 4:
+                *dListP = &D_060043D8;
+                break;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0E314.s")
+    if (limbIndex == 1) {
+        Matrix_MultVec3f(&sp7C, &this->unk_454);
+        Matrix_MultVec3f(&sp70, &this->unk_460);
+        Matrix_MultVec3f(&sp64, &this->unk_46C);
+        Matrix_MultVec3f(&sp58, &this->unk_478);
+        Matrix_MultVec3f(&sp4C, &this->unk_484);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0E430.s")
+    if (limbIndex == 5) {
+        Matrix_MultVec3f(&sp3C, &this->actor.focus);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0E5E0.s")
+    if (limbIndex == 4) {
+        gDPSetEnvColor(POLY_OPA_DISP++, this->unk_1F4.r, this->unk_1F4.g, this->unk_1F4.b, 0);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0E728.s")
+    Collider_UpdateSpheres(limbIndex, &this->collider);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_sw.c", 2145);
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0E90C.s")
+void EnSw_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dListP, Vec3s* rot, void* thisx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0E9BC.s")
+void func_80B0EDB8(GlobalContext* globalCtx, Color_RGBA8* arg1, s16 arg2, s16 arg3) {
+    f32 temp_f2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/EnSw_Update.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_sw.c", 2181);
+    temp_f2 = (11500.0f / arg3) * (arg3 - arg2);
+    if (0.0f == temp_f2) {
+        temp_f2 = 11500;
+    }
+    POLY_OPA_DISP = Gfx_SetFog2(POLY_OPA_DISP, arg1->r, arg1->g, arg1->b, arg1->a, 0, (s16)temp_f2);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_sw.c", 2197);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0EAC4.s")
+void func_80B0EEA4(GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0EDA4.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_sw.c", 2205);
+    POLY_OPA_DISP = func_800BC8A0(globalCtx, POLY_OPA_DISP);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_sw.c", 2207);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0EDB8.s")
+void EnSw_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnSw* this = THIS;
+    Color_RGBA8 sp30 = { 184, 0, 228, 255 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0EEA4.s")
+    if (((this->actor.params & 0xE000) >> 0xD) != 0) {
+        Matrix_RotateX(-1.3962634f, MTXMODE_APPLY);
```

Comment:
```suggestion
        Matrix_RotateX(-4 * M_PI / 9, MTXMODE_APPLY);
```
Assuming it matches. It's something like this at the very least.

Reply:
Assuming your math is right, possibly this:
```suggestion
        Matrix_RotateX(DEGF_TO_RADF(-80), MTXMODE_APPLY);
```